src/Polly.Core/Timeout/TimeoutStrategyOptions.cs

```diff
@@ -0,0 +1,32 @@
+using System.ComponentModel.DataAnnotations;
+using Polly.Builder;
+
+namespace Polly.Timeout;
+
+/// <summary>
+/// The options for the timeout strategy.
+/// </summary>
+public class TimeoutStrategyOptions : ResilienceStrategyOptions
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutStrategyOptions"/> class.
+    /// </summary>
+    public TimeoutStrategyOptions() => StrategyType = TimeoutConstants.StrategyType;
+
+    /// <summary>
+    /// Gets or sets the timeout generator that generates the timeout for a given execution.
+    /// </summary>
+    [Required]
+    public TimeoutGenerator TimeoutGenerator { get; set; } = new();
+
+    /// <summary>
+    /// Gets or sets the timeout event that notifies the timeout occurred.
+    /// </summary>
+    [Required]
+    public OnTimeoutEvent OnTimeout { get; set; } = new();
+
+    /// <summary>
+    /// Gets or sets the timeout strategy.
+    /// </summary>
```

Comment:
Please note what the default is for this property.

Reply:
I have dropped the support for pessimistic timeout in this PR to make it simpler. The property is no longer there.